### PR TITLE
[DFSM] Add INFO message for SharedStorage DeletionPolicy

### PR DESCRIPTION
**Pull request by chenwany**
Add INFO level of message when DeletionPolicy of shared storage is `Retain` or `Delete`
Signed-off-by: chenwany <chenwany@amazon.com>

### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/validators/cluster_validators.py
Comment (gmarciani) [inline cli/src/pcluster/validators/cluster_validators.py:755]: Generally speaking, I think we should not have warning message in both cases, but keep the one that we consider most important. I would vote for avoiding the message for the behaviour that we consider the default.

Comment (gmarciani) [reply cli/src/pcluster/validators/cluster_validators.py:755]: the other approach I agree with is to have INFO messages in both cases.

Comment (chenwany) [reply cli/src/pcluster/validators/cluster_validators.py:755]: Changed to INFO level

File: cli/src/pcluster/validators/cluster_validators.py
Comment (gmarciani) [inline cli/src/pcluster/validators/cluster_validators.py:758]: Minor: I would slightly rephrase to

```
The storage '{name}' will be deleted when you remove it from the configuration and perform a cluster update or a cluster delete.
```

Similarly to the below message for Retain

**Review by gmarciani — DISMISSED** [review]

**Review by hanwen-pcluste — APPROVED** [review]
Approve again after resolving merge conflicts

**Pull request merged**
